apepay/__init__.py

```diff
@@ -58,9 +58,44 @@ def normalize_address(cls, value: Any) -> AddressType:
 
     @property
     def contract(self) -> ContractInstance:
-        # TODO: Fix this
-        return self.project_manager.StreamManager.at(self.address)
-        # return self.chain_manager.contracts.instance_at(self.address)
+        try:
+            return self.chain_manager.contracts.instance_at(self.address)
+        except Exception:
+            return self.project_manager.StreamManager.at(self.address)
+
+    def _get_contract_receipt(self, start_block=0, stop_block=None) -> ReceiptAPI:
+        if stop_block is None:
+            stop_block = self.chain_manager.blocks.head.number
+
+        mid_block = (stop_block - start_block) // 2 + start_block
+        # NOTE: biased towards mid_block == start_block
+
+        if start_block == mid_block:
+            for tx in self.chain_manager.blocks[mid_block].transactions:
+                if tx.receipt.contract_address == self.address:
+                    return tx.receipt
+
+            raise  # cannot find receipt
```

Comment:
you can't do empty raises, that does not work in Python

Reply:
It'll work, but yeah it needs to be a better error (also I wanted to upstream this to Ape core as the backup method when you can't get the deployment receipt for a contract)

Reply:
no it will not raise!

Reply:
```
raise  # Must be done within an except block
raise Exception()  # This is probably what you are zintending to do here
```

Reply:
Can get rid of this anyways with https://github.com/ApeWorX/ape/pull/1548